libraries/AP_Scripting/examples/command_int.lua

```diff
@@ -0,0 +1,45 @@
+--[[
+   demonstrate using the gcs:command_int() interface to send commands from MAVLink MAV_CMD_xxx set
+--]]
+
+local MAV_FRAME_GLOBAL = 0
+
+local MAV_CMD_DO_SET_MODE = 176
+local MAV_CMD_DO_REPOSITION = 192
+
+-- some plane flight modes for testing
+local MODE_LOITER = 12
+local MODE_GUIDED = 15
+
+local MAV_MODE_FLAG_CUSTOM_MODE_ENABLED = 1
+
+--[[
+   test API calls. When in LOITER mode change to GUIDED and force flying to a location NE of home
+--]]
+local function test_command_int()
+   if vehicle:get_mode() ~= MODE_LOITER then
+      return
+   end
+   local home = ahrs:get_home()
+   if not home then
+      return
+   end
+
+   -- force mode GUIDED
+   gcs:command_int(MAV_FRAME_GLOBAL, MAV_CMD_DO_SET_MODE, { MAV_MODE_FLAG_CUSTOM_MODE_ENABLED, MODE_GUIDED })
+
+   -- and fly to 200m NE of home and 100m above home
+   local dest = home:copy()
+   dest:offset_bearing(45, 200)
+   dest:alt(home:alt()+100*100)
+
+   gcs:command_int(MAV_FRAME_GLOBAL, MAV_CMD_DO_REPOSITION, { -1, 0, 0, 0, dest:lat(), dest:lng(), dest:alt()*0.01 })
```

Comment:
The vehicle:update_target_location(old_location, new_location) method validates the old location hasn't changed before setting new location to "prevent a race condition where the user changes mode or commands a different target in the controlling lua script". Is that not a problem when using MAVLink?

Reply:
you can still use that method, this just offers access to a much broader range of functions, with the same semantics you get as over MAVLink from a companion computer.
If you need that race free update then you should use vehicle:update_target_location()